packages/plugin/src/Events/Notifications/NotificationResponseEvent.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+namespace Solspace\Freeform\Events\Notifications;
+
+use Psr\Http\Message\ResponseInterface;
+use Solspace\Freeform\Events\CancelableArrayableEvent;
+use Solspace\Freeform\Library\Notifications\AbstractNotification;
+
+class NotificationResponseEvent extends CancelableArrayableEvent
+{
+    private AbstractNotification $notification;
+
+    private ResponseInterface $response;
+
+    public function __construct(AbstractNotification $notification, ResponseInterface $response)
+    {
+        $this->notification = $notification;
+        $this->response = $response;
+
+        parent::__construct();
+    }
```

Comment:
With [PHP8 there's a neat little improvement](https://stitcher.io/blog/constructor-promotion-in-php-8) where you can declare the variables and their visibility directly in the constructor

```suggestion
    public function __construct(
        private AbstractNotification $notification, 
        private ResponseInterface $response
    ) {
        parent::__construct();
    }
```